src/bootchooser.c

```diff
@@ -1011,18 +1012,22 @@ static RaucSlot *efi_get_primary(GError **error) {
 		goto out;
 	}
 
-	bootnext = g_list_nth_data(bootorder_entries, 0);
+	bootnext = g_list_nth_data(bootorder_entries, idx++);
 
 get_slot:
 
-	/* find matching slot entry */
-	g_hash_table_iter_init(&iter, r_context()->config->slots);
-	while (g_hash_table_iter_next(&iter, NULL, (gpointer *)&slot)) {
-		if (g_strcmp0(bootnext->name, slot->bootname) == 0) {
-			primary = slot;
-			break;
+	do {
+		/* find matching slot entry */
+		g_hash_table_iter_init(&iter, r_context()->config->slots);
+		while (g_hash_table_iter_next(&iter, NULL, (gpointer *)&slot)) {
+			if (g_strcmp0(bootnext->name, slot->bootname) == 0) {
+				primary = slot;
+				break;
+			}
 		}
-	}
+
+		bootnext = g_list_nth_data(bootorder_entries, idx++);
```

Comment:
We should be able to just use g_list_next() for iteration, which seems more ideomatic.

Reply:
I obtain `bootnext` from `efi_bootorder_get()` and thus use g_list_nth_data() to directly get the an alternative for `bootnext` if `BootNext` EFI var was not set.
As I cannot run g_list_next() on the data object, I chose to iterate here this way.

Otherwise I have to change to code to use a GList for iterating and add some more extra handling to use the `efi_bootentry` `bootnext` I get from `efi_bootorder_get()`.